EXILED/Exiled.API/Features/Player.cs

```diff
@@ -588,9 +588,18 @@ public PlayerPermissions RemoteAdminPermissions
         public Role Role
         {
             get => role ??= Role.Create(RoleManager.CurrentRole);
-            internal set => role = value;
+            internal set
+            {
+                PreviousRole = role;
+                role = value;
+            }
         }
 
+        /// <summary>
+        /// Gets the role that player had before changing role.
+        /// </summary>
+        public Role PreviousRole { get; private set; }
```

Comment:
I don't know how I feel about this. It will break if for example the PreviousRole was 106, and you try to change the AHP. Could you check that?

Also, it will take memory until it gets dereferenced.

Reply:
It can be useful in some cases. I can make property such as `IsDisabled` or something else that will block setting values for role. I thought about setting just RoleTypeId instead of `Role` but it seems to me that storing `Role` is more useful in more cases

Reply:
> I don't know how I feel about this. It will break if for example the PreviousRole was 106, and you try to change the AHP. Could you check that?
> 
> Also, it will take memory until it gets dereferenced.

It's actually fine it's clearly said PreviousRole if people do this it's a skll issue from dev 

For the memory it's will just be stop when event end ?

Reply:
> > I don't know how I feel about this. It will break if for example the PreviousRole was 106, and you try to change the AHP. Could you check that?
> > 
> > Also, it will take memory until it gets dereferenced.
> 
> It's actually fine it's clearly said PreviousRole if people do this it's a skll issue from dev 

Also yeah, name says that player doesn't have this role anymore and it's just cached

> For the memory it's will just be stop when event end ?

Which event? Role is stored until other role is set. Events aren't related to this

Reply:
Actually, role type may be smarter to use. Using the role can be a pain in the ass, and more if it is pooled and reused with another player.